### PR TITLE
fix for 543 and added PUT tests

### PR DESCRIPTION
**Pull request by himaninare**

### CODE REVIEW

File: src/test/java/com/ccs/conclave/api/cii/tests/PostSchemeInfoTests.java [hunk 1]
```diff
@@ -173,16 +173,17 @@ public void postSchemeInfoWithInvalidPrimaryID() {
...
+    @Test //CON-543 - Fixed
     public void postSchemeInfoWithNoAdditionalIdentifierSection() {
         SchemeInfo schemeInfo = OrgDataProvider.getInfo(NORTHERN_CHARITY_WITH_COH);
+        String responseStr = getSchemeInfoWithoutAddIdentifierSection(schemeInfo, NORTHERN_CHARITY_WITH_COH);
 
         //Modify the response to Remove The Additional Identifier Section
-        String responseStr = getSchemeInfoWithoutAddIdentifierSection(schemeInfo, NORTHERN_CHARITY_WITH_COH);
-        Response response = RestRequests.postSchemeInfo(responseStr);
-        //CON-543: Currently we get 500 but expected to register successfully an return ccs-org-id
         SchemeInfo expectedSchemeInfo = getInfoWithoutAddIdentifiers(NORTHERN_CHARITY_WITH_COH);
```
Comment (amrutha-sireesh) [inline src/test/java/com/ccs/conclave/api/cii/tests/PostSchemeInfoTests.java:182]: Please move this above Post operation

File: src/main/java/com/ccs/conclave/api/cii/data/OrgDataProvider.java [hunk 2]
```diff
@@ -489,8 +506,9 @@ public static String getSchemeInfoWithInvalidPrimaryID(SchemeInfo schemeInfo, Sc
     public static String getSchemeInfoWithoutAddIdentifierSection(SchemeInfo schemeInfo, SchemeRegistry scheme) {
         Response response = RestRequests.getSchemeInfo(scheme, schemeInfo.getIdentifier().getId());
         verifyGetSchemeInfoResponse(schemeInfo, response); // verify Get SchemeInfo response before using it
-        String[] strings = response.asString().split("additionalIdentifiers\":\\[(.*?)\\],");
-        String responseWithoutAddIdentifierSection = strings[0] + strings[1];
+        String[] strings = response.asString().split("additionalIdentifiers\":\\[(.*?)\\]");
```
Comment (amrutha-sireesh) [inline src/main/java/com/ccs/conclave/api/cii/data/OrgDataProvider.java:509]: can you please try regex
response.asString().split("additionalIdentifiers\":\\[(.*?)\\],\"");

File: src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java [hunk 3]
```diff
@@ -74,19 +76,77 @@ public void updateScheme_COH_and_CHC_into_SC() {
...
+        logger.info("Successful post operation...");
+
+        logger.info("Adding additional identifier to the existing organisation...");
+        AdditionalSchemeInfo additionalSchemeInfo = additionalSchemesInfo.get(0);
+        response = RestRequests.updateScheme("7310710000000", additionalSchemeInfo);
+        verifyInvalidIdResponse(response);
+    }
 
-    // Todo @Test
+    @Test
```
Comment (amrutha-sireesh) [inline src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java:127]: Can you please refer the explanation in CON-616 as a NOTE above this test

File: src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java [hunk 4]
```diff
@@ -74,19 +76,77 @@ public void updateScheme_COH_and_CHC_into_SC() {
...
     public void updateScheme_InvalidOrgId() {
+        SchemeInfo schemeInfo = OrgDataProvider.getInfo(CHARITIES_COMMISSION_WITH_SC);
+        // GetScheme response without additional identifiers
+        String responseStr = getSchemeInfoWithEmptyAddIdentifiers(schemeInfo, CHARITIES_COMMISSION_WITH_SC);
+        // Get expected SchemeInfo without additional identifiers
+        SchemeInfo expectedSchemeInfo = getInfoWithoutAddIdentifiers(CHARITIES_COMMISSION_WITH_SC);
+        // get only AdditionalIdentifiers from the given Scheme
+        List<AdditionalSchemeInfo> additionalSchemesInfo = getAdditionalIdentifierInfo(INVALID_SCHEME);
```
Comment (amrutha-sireesh) [inline src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java:111]: Please pass the valid Scheme as this test is for invalid OrgId

File: src/main/java/com/ccs/conclave/api/cii/data/OrgDataProvider.java (hunk 2 above)
Comment (himaninare) [reply src/main/java/com/ccs/conclave/api/cii/data/OrgDataProvider.java:509]: Nope It did not work

File: src/test/java/com/ccs/conclave/api/cii/tests/PostSchemeInfoTests.java (hunk 1 above)
Comment (himaninare) [reply src/test/java/com/ccs/conclave/api/cii/tests/PostSchemeInfoTests.java:182]: Already Done

File: src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java (hunk 4 above)
Comment (himaninare) [reply src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java:111]: Done

File: src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java (hunk 3 above)
Comment (himaninare) [reply src/test/java/com/ccs/conclave/api/cii/tests/UpdateSchemeTests.java:127]: Done

File: src/main/java/com/ccs/conclave/api/cii/data/OrgDataProvider.java (hunk 2 above)
Comment (himaninare) [reply src/main/java/com/ccs/conclave/api/cii/data/OrgDataProvider.java:509]: Done

**Review by amrutha-sireesh — APPROVED** [review]

**Pull request merged**